### ISSUE TITLE
1y fees / liquidity appears to be calculated incorrectly

### ISSUE THREAD

**Issue by LincolnCannon** [pangolindex/analytics-avalanche#7]
Issue Location: https://info.pangolin.exchange/#/pairs

In this report, the "1y fees / liquidity" column appears to be calculated incorrectly.

If you look at the same UI for Uniswap, they're calculated correctly based on (24hr fees * 365) / liquidity. If these Pangolin numbers were calculated the same way, the numbers in this column would be much lower.

**Comment by HariSeldon23** [comment]
If you look here https://github.com/pangolindex/analytics/blob/bed42fd0b00e1fae43a6573d1cbb91b751989356/src/components/PairList/index.js

Then you'll see the following on line 155
`const apy = formattedPercent((pairData.oneDayVolumeUSD * 0.003 * 365 * 100) / pairData.reserveUSD)`

So looking at that the calculation is 1 day USD Volume * fee's percentage * 365 days a year divided by the reserve USD figure. 

The reserveUSD is then the total amount locked into the Liquidity Pool. 

So I do think you're correct that it appears wrong. If we look at the below for WAVAX-USDT
![image](https://user-images.githubusercontent.com/3041005/109074514-efcb0400-769b-11eb-98f7-0dcb7ef0fd34.png)

We can then perform the calculation detailed and get severely different figures
![image](https://user-images.githubusercontent.com/3041005/109074795-489a9c80-769c-11eb-9f14-4a803840a032.png)

**Comment by LincolnCannon** [comment]
Why should there be a multiplication of 100 in the numerator? It seems like
removing that would fix the problem.

**Comment by bmino** [comment]
We are indeed calculating using this formula:
```const apy = formattedPercent((pairData.oneDayVolumeUSD * 0.003 * 365 * 100) / pairData.reserveUSD)```

Which is correct... if those variables meant what their naming convention leads you to believe.
That `pairData.reserveUSD` is actually given in terms of AVAX and not USD. The other values in this formula use USD values, so this also needs to be converted using the following:

```const apy = formattedPercent((pairData.oneDayVolumeUSD * 0.003 * 365 * 100) / (pairData.reserveUSD * priceEth))```

Note that I used `priceEth` which is actually the price of AVAX. Confused yet haha?

Even better though! We actually do this logic already in `PairData.js` and store the appropriate result in `trackedReserveUSD` and I propose we use this instead:

```
const apy = formattedPercent((pairData.oneDayVolumeUSD * 0.003 * 365 * 100) / pairData.trackedReserveUSD)
```

**Comment by LincolnCannon** [comment]
Why should the formula for this column on Pangolin be different than on Uniswap?

**Comment by bmino** [comment]
The "formula" is the same. The problem is that on uniswap everything is in terms of ETH, but here values are in terms of AVAX and some refactoring is needed to fully integrate that. Pangolin isn't just a copy pasta of uniswap

**Comment by LincolnCannon** [comment]
The column header on both Pangolin and Uniswap says "1y fees / liquidity". On Uniswap, you can use the "fees (24hr)" multiplied by 365 and divided by the "liquidity" column to verify the "1y fees / liquidity" column. You can't do that on Pangolin. If your calculation is desirable then, in the least, the column header should be renamed to reflect your calculation rather than miscommunicate the content of the column.

**Comment by bmino** [comment]
The columns on Pangolin for `Liquidity` and `Fees (24hr)` are correct. The issue you are experiencing is that the `1y Fees / Liquidity` column is NOT an accurate representation of non-compounding APY which is what dividing yearly fees by liquidity should represent.

I mean you could rename that column to "1y Fees (USD) / Liquidity (AVAX)" which is the currently displayed value but that's useless right?
My proposed fix is to calculate that column value exactly as the title says

**Comment by LincolnCannon** [comment]
The "liquidity" column uses the USD symbol in its values. It's reasonable to assume that the reference to "liquidity" in "1y fees / liquidity" is a reference to the same value that's displayed in the "liquidity" column. In fact, any other assumption is unreasonable contextually. I believe your proposal would perpetuate confusion.

**Comment by bmino** [comment]
Okay I think I'm seeing what you are saying. Correct me if I'm wrong here.

The current value calculation is:
```
const liquidity = formattedNum(pairData.reserveUSD * ethPrice, true)
const volume = formattedNum(pairData.oneDayVolumeUSD, true)
const apy = formattedPercent((pairData.oneDayVolumeUSD * 0.003 * 365 * 100) / pairData.reserveUSD)
```

and I proposed changing the `apy` calculation to use `pairData.trackedReserveUSD` and your concern is that now we use one method (`pairData.reserveUSD * ethPrice`) to calculate `liquidity` and another (`pairData.trackedReserveUSD`) to represent liquidity when calculating `apy`?

If so, then yes I agree and this should instead become:

```
const liquidity = formattedNum(pairData.trackedReserveUSD, true)
const volume = formattedNum(pairData.oneDayVolumeUSD, true)
const apy = formattedPercent((pairData.oneDayVolumeUSD * 0.003 * 365 * 100) / pairData.trackedReserveUSD)
```

**Comment by dasconnor** [comment]
The biggest problem I ran into was that I didn't have stablecoins when I was developing this. I couldn't get any prices on-chain, so I had to price everything in terms of AVAX instead of USD. Most USD values are actually in terms of AVAX where the price of AVAX = $1, then I just multiplied by the `ethPrice` (which is actually the avax price) which I determine from querying coin gecko's api. I think this fix should work tho since we're only tracking a percentage change.
